tagstudio/src/qt/ts_qt.py

```diff
@@ -948,6 +953,40 @@ def select_item(self, grid_index: int, append: bool, bridge: bool):
     def set_macro_menu_viability(self):
         self.autofill_action.setDisabled(not self.selected)
 
+    def update_completions_list(self, text: str) -> None:
+        matches = re.search(r"(mediatype|filetype|path|tag_id|tag):(\"?[A-Za-z0-9\ \t]+\"?)?", text)
+        if not matches:
+            return
+
+        query_type: str
+        query_value: str | None
+        query_type, query_value = matches.groups()
+
+        completion_list: list[str] = []
+
+        if query_type == "tag":
+            completion_list = list(map(lambda x: "tag:" + x.name, self.lib.tags))
+        elif query_type == "tag_id":
+            completion_list = list(map(lambda x: "tag_id:" + str(x.id), self.lib.tags))
+        elif query_type == "path":
+            completion_list = list(map(lambda x: "path:" + x, self.lib.get_paths()))
+        elif query_type == "mediatype":
+            completion_list = list(
+                map(lambda x: 'mediatype:"' + x.name + '"', MediaCategories.ALL_CATEGORIES)
+            )
```

Comment:
I have added this, but the reason I didn't add it originally is because when you type `mediatype:` and scroll through the completions, you will get the section that isn't quoted and then the section that is quoted, just due to how the QCompleter works. Niche and, now that I think about it, probably not an issue, but a little quirky.

Actually now that I'm trying it, it is even less of an issue because you first have to type `mediatype: ` (with a space) for it to fill in the completion list with all of the mediatypes, and then remove the space and scroll through.

Reply:
So is there no (straightforward) way to just have the suggestions with quotations show up using QCompleter?

Reply:
As far as I understand it, no. If you look at [QCompleter.setFilterMode()](https://doc.qt.io/qt-6/qcompleter.html#filterMode-prop), it uses [Qt::MatchFlags](https://doc.qt.io/qt-6/qt.html#MatchFlag-enum) to define how it decides what items to show from its completion list, but the description of setFilterMode() tells you that only MatchStartsWith, MatchContains, and MatchEndsWith are implemented. Using anything else just throws an error.

Because of that, the unquoted and quoted mediatypes will never show up at the same time, because the quoted ones technically start with `mediatype:"` and the unquoted ones start with `mediatype:`. My hack around that was to just always have them quoted, but that is annoying and, I'm now realizing, will not work for single quotes unless we explicitly add it.

Just as a note: we can't use MatchContains, because, for the same reason as above, the completion list needs to have items that start with the type of query (i.e. we can't just have a list that contains `photoshop`, `plaintext`, etc. They need to be `mediatype:photoshop`, `mediatype:plaintext`, or `mediatype:"photoshop"`, `mediatype:"plaintext"`).

Reply:
Dang, sounds like it would involve quite a bit of custom work in order to do this. Thanks for looking into this, perhaps it's something we can tackle in the future, but for now I'm content settling for Qt's limitations here if you are.

Reply:
Yeah, as it stands it is pretty far from what I was envisioning, but still useful.

Reply:
Sounds like we're on the same page, works enough for this PR then 👍